src/crystal/system/win32/file_descriptor.cr

```diff
@@ -34,20 +24,25 @@ module Crystal::System::FileDescriptor
     end
   end
 
+  private def unbuffered_read_blocking(handle, slice)
+    ret = LibC.ReadFile(handle, slice, slice.size, out bytes_read, nil)
+    return bytes_read.to_i32 unless ret.zero?
+
+    case error = WinError.value
+    when .error_access_denied?
+      raise IO::Error.new "File not open for reading", target: self
+    when .error_broken_pipe?
+      return 0_i32
+    else
+      raise IO::Error.from_os_error("Error reading file", error, target: self)
+    end
+  end
+
   private def unbuffered_write(slice : Bytes) : Nil
     handle = windows_handle
     until slice.empty?
       if system_blocking?
-        if LibC.WriteFile(handle, slice, slice.size, out bytes_written, nil) == 0
-          case error = WinError.value
-          when .error_access_denied?
-            raise IO::Error.new "File not open for writing", target: self
-          when .error_broken_pipe?
-            return 0_u32
```

Comment:
Actually, this early return from the `until` loop is now gone

Reply:
Oops. Let's revert this then.